src/EventAction.cpp

```diff
@@ -41,81 +40,88 @@ void EventAction::BeginOfEventAction(const G4Event*)
 }
 
 
-void EventAction::EndOfEventAction(const G4Event* event)
+void EventAction::EndOfEventAction(const G4Event* g4event)
 {
+    G4int event_id_offset_ = ConfigManager::GetEventIDOffset();
+    G4double energy_threshold_ = ConfigManager::GetEnergyThreshold();
+
     // get MC truth manager
     MCTruthManager * mc_truth_manager = MCTruthManager::Instance();
 
     // get map of particles from MC truth manager
-    auto const MCParticleMap = mc_truth_manager->GetMCParticleMap();
+    const MCTruthManager::MCParticleMap& particleMap = mc_truth_manager->GetMCParticleMap();
+
 
     double energy_deposited = 0.;
 
     // add particle to analysis manager
-    for (auto const& p : MCParticleMap)
+    for (auto const& p : particleMap)
     {
-        auto const& particle = p.second;
+        const MCParticle* particle = p.second;
         energy_deposited += particle->EnergyDeposited();
         // std::cout << "Energy deposited by particle PDG (" << particle->PDGCode() << "): " << particle->EnergyDeposited() << std::endl;
     }
 
-    int mod = event->GetEventID() % 1000;
-    if (mod == 0)
-    {
-        G4cout << "Event " << event->GetEventID() << "..." << G4endl;
+    //Replace the following lines with "/run/printProgress 1000" in the macro file
+    //int mod = g4event->GetEventID() % 1;
+    //if (mod == 0)
+    //{
+    //    G4cout << "Event " << g4event->GetEventID() << "..." << G4endl;
         // G4cout << "Energy threshold: " << energy_threshold_ << G4endl;
         // G4cout << "Total energy deposited: " << energy_deposited << G4endl;
```

Comment:
Don't leave commented out code in the repository.

Reply:
fixing in bugfix/v3.0.0_revisions branch